src/Tickets/Commerce/Status/Status_Abstract.php

```diff
@@ -148,6 +148,29 @@ public function can_apply_to( $order, $new_status ) {
 		return true;
 	}
 
+	/**
+	 * Whether a Status Interface can be applied on top of another Status Interface.
+	 *
+	 * @since TBD
+	 *
+	 * @param self $new_status The new status.
+	 * @param ?int $order_id   Which order we are testing against.
+	 *
+	 * @return bool Whether the new status can be applied to the current status.
+	 */
+	public function can_transition_to_status( $new_status, $order_id = null ) {
+		if ( $this->get_wp_slug() === $new_status->get_wp_slug() ) {
+			// Allow from refunded to refunded in order to support multiple refunds.
+			return 'refunded' === $this->get_slug();
```

Comment:
Shouldn't this really live in the Refunded status class?
The abstract class should have no knowledge if its children. The children should override this value.

Reply:
Move that and this whole function becomes
`return $this->is_final()`

Reply:
this was in abstact already - fixing that for my changes wont do it for existing code.